internal/backend/projects/lint.go

```diff
@@ -0,0 +1,371 @@
+
+var (
+	lintCheckers []Checker
+)
+
+const manifestFile = "autokitteh.yaml"
+
+func init() {
+	lintCheckers = []Checker{
+		// Generic
+		checkNoTriggers,
+		checkEmptyVars,
+		checkSize,
+		checkConnectionNames,
+		checkTriggerNames,
+
+		// Runtime
+		checkHandlers,
+		checkCodeConnections,
+	}
+}
+
+func Validate(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation {
+	var vs []*sdktypes.CheckViolation
+	for _, checker := range lintCheckers {
+		vs = append(vs, checker(projectID, manifest, resources)...)
+	}
+
+	return vs
+}
+
+var Rules = map[string]string{ // ID -> Descrption
+	"E1": "No triggers defined",
+	"E2": "Project size too large",
+	"E3": "Duplicate connection name",
+	"E4": "Duplicate trigger name",
+	"E5": "Bad `call` format",
+	"E6": "File not found",
+	"E7": "Syntax error",
+	"E8": "Missing handler",
+	"E9": "Nonexisting connection",
+
+	"W1": "Empty variable",
+}
+
+func checkNoTriggers(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation {
+	if len(m.Project.Triggers) == 0 {
+		return []*sdktypes.CheckViolation{
+			{
+				FileName: manifestFile,
+				Level:    sdktypes.ViolationError,
+				Message:  "no triggers",
+				RuleId:   "E1",
+			},
+		}
+	}
+
+	return nil
+}
+
+func checkEmptyVars(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation {
+	var vs []*sdktypes.CheckViolation
+	for _, v := range m.Project.Vars {
+		if v.Value == "" {
+			vs = append(vs, &sdktypes.CheckViolation{
+				FileName: manifestFile,
+				Level:    sdktypes.ViolationWarning,
+				Message:  fmt.Sprintf("variable %q is empty", v.Name),
+				RuleId:   "W1",
+			})
+		}
+	}
+
+	for _, conn := range m.Project.Connections {
+		for _, v := range conn.Vars {
+			if v.Value == "" {
+				vs = append(vs, &sdktypes.CheckViolation{
+					FileName: manifestFile,
+					Level:    sdktypes.ViolationWarning,
+					Message:  fmt.Sprintf("connection %q variable %q is empty", conn.Name, v.Name),
+					RuleId:   "W1",
+				})
+			}
+		}
+	}
+
+	return vs
+}
+
+const (
+	mb = 1 << 20
+)
+
+func checkSize(_ sdktypes.ProjectID, _ *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation {
+	maxProjectSize := projectsgrpcsvc.Configs.Default.MaxUploadSize
+
+	total := 0
+	for _, data := range resources {
+		total += len(data)
+	}
+
+	if total > maxProjectSize {
+		sizeMB := float64(total) / mb
+		return []*sdktypes.CheckViolation{
+			{
+				FileName: manifestFile,
+				Level:    sdktypes.ViolationError,
+				Message:  fmt.Sprintf("project size (%.2fMB) exceeds limt of %dMB", sizeMB, maxProjectSize/mb),
+				RuleId:   "E2",
+			},
+		}
+	}
+
+	return nil
+}
+
+func checkConnectionNames(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation {
+	names := make(map[string]int) // name -> count
```

Comment:
when checking names, we need to make sure it paths validation through `sdktypes.ParseSymbol`.

Reply:
also for triggers, projects, etc...